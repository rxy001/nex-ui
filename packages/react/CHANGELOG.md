# @nex-ui/react

## 0.0.15

### Patch Changes

- fdd4483: Release
- Updated dependencies [fdd4483]
  - @nex-ui/styled@0.0.15
  - @nex-ui/system@0.0.15
  - @nex-ui/hooks@0.0.15
  - @nex-ui/icons@0.0.15
  - @nex-ui/utils@0.0.15

## 0.0.14

### Patch Changes

- ca25983: Release
- Updated dependencies [ca25983]
  - @nex-ui/styled@0.0.14
  - @nex-ui/system@0.0.14
  - @nex-ui/hooks@0.0.14
  - @nex-ui/icons@0.0.14
  - @nex-ui/utils@0.0.14

## 0.0.13

### Patch Changes

- 1b97788: Release
- Updated dependencies [1b97788]
  - @nex-ui/styled@0.0.13
  - @nex-ui/system@0.0.13
  - @nex-ui/hooks@0.0.13
  - @nex-ui/icons@0.0.13
  - @nex-ui/utils@0.0.13

## 0.0.12

### Patch Changes

- 664eb21: Test

## 0.0.11

### Patch Changes

- 259356e: Test
- Updated dependencies [259356e]
  - @nex-ui/styled@0.0.11
  - @nex-ui/system@0.0.11
  - @nex-ui/hooks@0.0.11
  - @nex-ui/icons@0.0.11
  - @nex-ui/utils@0.0.10

## 0.0.10

### Patch Changes

- 5a229bd: release
- Updated dependencies [5a229bd]
  - @nex-ui/styled@0.0.10
  - @nex-ui/system@0.0.10
  - @nex-ui/icons@0.0.10

## 0.0.9

### Patch Changes

- 4a5c359: release
- Updated dependencies [4a5c359]
  - @nex-ui/styled@0.0.9
  - @nex-ui/system@0.0.9
  - @nex-ui/hooks@0.0.9
  - @nex-ui/icons@0.0.9
  - @nex-ui/utils@0.0.9

## 0.0.8

### Patch Changes

- c8c1956: release
- Updated dependencies [c8c1956]
  - @nex-ui/styled@0.0.8
  - @nex-ui/system@0.0.8
  - @nex-ui/hooks@0.0.8
  - @nex-ui/icons@0.0.8
  - @nex-ui/utils@0.0.8

## 0.0.7

### Patch Changes

- 2618e1a: release
- Updated dependencies [2618e1a]
  - @nex-ui/styled@0.0.7
  - @nex-ui/system@0.0.7
  - @nex-ui/hooks@0.0.7
  - @nex-ui/icons@0.0.7
  - @nex-ui/utils@0.0.7

## 0.0.6

### Patch Changes

- 87191c1: release
- Updated dependencies [87191c1]
  - @nex-ui/styled@0.0.6
  - @nex-ui/system@0.0.6
  - @nex-ui/hooks@0.0.6
  - @nex-ui/icons@0.0.6
  - @nex-ui/utils@0.0.6

## 0.0.5

### Patch Changes

- 7631577: release
- Updated dependencies [7631577]
  - @nex-ui/styled@0.0.5
  - @nex-ui/system@0.0.5
  - @nex-ui/hooks@0.0.5
  - @nex-ui/icons@0.0.5
  - @nex-ui/utils@0.0.5

## 0.0.4

### Patch Changes

- 70cb478: v0.0.4
- Updated dependencies [70cb478]
  - @nex-ui/styled@0.0.4
  - @nex-ui/system@0.0.4
  - @nex-ui/hooks@0.0.4
  - @nex-ui/icons@0.0.4
  - @nex-ui/utils@0.0.4
