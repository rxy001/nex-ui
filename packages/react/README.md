<div style="background:white;">
  <p align="center">
    <img src='../../assets/logo.png' style="width: 150px; height: 150px;" />
  </p>
  <h1 align="center">Nex UI</h1>
  <p align="center">ðŸŽ‰ A beautiful, modern, and reliable React component library.</p>
  <p align="center"><b>Theme Customizable, Light & Dark UI, Unique DX, Written In TypeScript</b></p>
</div>

## Documentation

Visit https://nex-ui-docs.vercel.app to view the full documentation.

## Getting Started

Visit https://nex-ui-docs.vercel.app/docs/getting-started to get started with Nex UI.

## Storybook

Visit https://nex-ui-storybook.vercel.app to view the storybook for all components.

## Contributing

Contributions are always welcome!

See [CONTRIBUTING.md](/CONTRIBUTING.md) for ways to get started.

Please adhere to this project's [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md).

## License

[MIT](https://choosealicense.com/licenses/mit/)
