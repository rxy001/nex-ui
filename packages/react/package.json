{
  "name": "@nex-ui/react",
  "version": "0.0.12",
  "author": "X1ng Yu",
  "description": "ðŸŽ‰ A beautiful, modern, and reliable React component library.",
  "homepage": "https://nex-ui-docs.vercel.app/",
  "private": false,
  "keywords": [
    "react",
    "react-component",
    "nui",
    "nex-ui"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/rxy001/nex-ui.git",
    "directory": "packages/react"
  },
  "license": "MIT",
  "scripts": {
    "build": "tsx ../../scripts/build.ts --dts",
    "build:fast": "tsx ../../scripts/build.ts",
    "typecheck": "tsc --noEmit",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "src/index.ts",
  "devDependencies": {
    "@storybook/react": "8.6.12"
  },
  "dependencies": {
    "@nex-ui/hooks": "workspace:*",
    "@nex-ui/icons": "workspace:*",
    "@nex-ui/styled": "workspace:*",
    "@nex-ui/system": "workspace:*",
    "@nex-ui/utils": "workspace:*",
    "clsx": "2.1.1"
  },
  "peerDependencies": {
    "@emotion/react": ">=11",
    "motion": "^12.7.2",
    "react": ">=19",
    "react-dom": ">=19"
  },
  "sideEffects": false,
  "clean-package": "../../clean-package.config.json"
}
