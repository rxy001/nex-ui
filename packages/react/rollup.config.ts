import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { defineConfig } from 'rollup'
import typescript from '@rollup/plugin-typescript'
import del from 'rollup-plugin-delete'
import dts from 'rollup-plugin-dts'
import { nexUIRollupPlugin } from '@nex-ui/plugins'

const dirname = fileURLToPath(new URL('./', import.meta.url))

export default defineConfig([
  {
    plugins: [
      del({
        targets: './dist/*',
      }),
      typescript({
        tsconfig: path.resolve(dirname, 'tsconfig.json'),
      }),
      nexUIRollupPlugin(),
    ],
    input: './src/index.ts',
    external: [
      'react',
      'react-dom',
      'react/jsx-runtime',
      'react-dom/client',
      '@nex-ui/utils',
      '@nex-ui/css-system',
      'classnames',
      'framer-motion',
      '@iconify/react',
      '@vanilla-extract/dynamic',
      '@vanilla-extract/css',
      '@vanilla-extract/css/functionSerializer',
    ],
    output: [
      {
        preserveModules: true,
        format: 'es',
        dir: './dist/es',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
      {
        preserveModules: true,
        format: 'cjs',
        dir: './dist/lib',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
    ],
  },
  {
    plugins: [dts()],
    input: './src/index.ts',
    output: {
      dir: './dist/types',
      format: 'es',
      preserveModules: true,
    },
  },
])
