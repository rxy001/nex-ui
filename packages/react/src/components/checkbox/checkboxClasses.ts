import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const checkboxClasses = generateUtilityClasses('nui-checkbox', [
  // ---------root---------
  'root',

  // "sm" | "md" | "lg" | "full" | "none"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  // ---------root---------

  'input',
  'label',
  'icon',
  'iconContainer',
])
