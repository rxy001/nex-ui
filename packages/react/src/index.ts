import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
