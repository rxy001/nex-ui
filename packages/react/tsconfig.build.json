{
  "extends": "../../tsconfig.json",
  "include": ["./src"],
  "exclude": ["./src/**/__tests__/**/*", "./src/**/__stories__/**/*"]
}
