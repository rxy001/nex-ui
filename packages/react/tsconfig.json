{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "~/*": ["../../*"],
      "@theme": ["./src/theme"]
    }
  },
  "include": ["./src", "./rollup.config.ts"]
}
