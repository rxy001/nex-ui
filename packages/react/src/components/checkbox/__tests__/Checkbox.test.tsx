import { createRef } from 'react'
import { fireEvent, act } from '@testing-library/react'
import {
  mountTest,
  renderWithNexUIProvider,
  rootClassNameTest,
} from '~/tests/shared'
import { Checkbox, CheckboxGroup } from '../index'
import { checkboxClasses } from '../checkboxClasses'

describe('Checkbox', () => {
  mountTest(<Checkbox />, {
    useAct: true,
  })

  rootClassNameTest(<Checkbox className='test-class' />, 'test-class', {
    useAct: true,
  })

  it("should forward ref to Checkbox's input element", async () => {
    const ref = createRef<HTMLInputElement>()
    const { getByRole } = await renderWithNexUIProvider(
      <Checkbox ref={ref} />,
      {
        useAct: true,
      },
    )
    const checkbox = getByRole('checkbox')
    expect(ref.current).toBe(checkbox)
  })

  it("should render Checkbox's label with text children", async () => {
    const { getByText } = await renderWithNexUIProvider(
      <Checkbox>Test Label</Checkbox>,
      {
        useAct: true,
      },
    )
    expect(getByText('Test Label')).toBeInTheDocument()
  })

  it('should render Checkbox with the root, size-md, radius-md, and color-blue classes but no others', async () => {
    const { container } = await renderWithNexUIProvider(<Checkbox />, {
      useAct: true,
    })

    const checkbox = container.firstElementChild
    expect(checkbox).toHaveClass(checkboxClasses.root)
    expect(checkbox).toHaveClass(checkboxClasses['size-md'])
    expect(checkbox).toHaveClass(checkboxClasses['radius-md'])
    expect(checkbox).toHaveClass(checkboxClasses['color-blue'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-green'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-cyan'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-orange'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-pink'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-purple'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-yellow'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-red'])
    expect(checkbox).not.toHaveClass(checkboxClasses['color-gray'])
    expect(checkbox).not.toHaveClass(checkboxClasses['size-sm'])
    expect(checkbox).not.toHaveClass(checkboxClasses['size-lg'])
    expect(checkbox).not.toHaveClass(checkboxClasses['radius-sm'])
    expect(checkbox).not.toHaveClass(checkboxClasses['radius-lg'])
    expect(checkbox).not.toHaveClass(checkboxClasses['radius-full'])
    expect(checkbox).not.toHaveClass(checkboxClasses.indeterminate)
    expect(checkbox).not.toHaveClass(checkboxClasses.checked)
    expect(checkbox).not.toHaveClass(checkboxClasses.disabled)
  })

  it('should add the appropriate color class to root element based on color prop', async () => {
    const { getByTestId } = await renderWithNexUIProvider(
      <>
        <Checkbox color='red' data-testid='color-red'>
          Checkbox
        </Checkbox>
        <Checkbox color='blue' data-testid='color-blue'>
          Checkbox
        </Checkbox>
        <Checkbox color='cyan' data-testid='color-cyan'>
          Checkbox
        </Checkbox>
        <Checkbox color='orange' data-testid='color-orange'>
          Checkbox
        </Checkbox>
        <Checkbox color='pink' data-testid='color-pink'>
          Checkbox
        </Checkbox>
        <Checkbox color='purple' data-testid='color-purple'>
          Checkbox
        </Checkbox>
        <Checkbox color='gray' data-testid='color-gray'>
          Checkbox
        </Checkbox>
        <Checkbox color='yellow' data-testid='color-yellow'>
          Checkbox
        </Checkbox>
        <Checkbox color='green' data-testid='color-green'>
          Checkbox
        </Checkbox>
      </>,
      {
        useAct: true,
      },
    )

    expect(getByTestId('color-red').parentElement).toHaveClass(
      checkboxClasses['color-red'],
    )
    expect(getByTestId('color-blue').parentElement).toHaveClass(
      checkboxClasses['color-blue'],
    )
    expect(getByTestId('color-cyan').parentElement).toHaveClass(
      checkboxClasses['color-cyan'],
    )
    expect(getByTestId('color-orange').parentElement).toHaveClass(
      checkboxClasses['color-orange'],
    )
    expect(getByTestId('color-pink').parentElement).toHaveClass(
      checkboxClasses['color-pink'],
    )
    expect(getByTestId('color-purple').parentElement).toHaveClass(
      checkboxClasses['color-purple'],
    )
    expect(getByTestId('color-green').parentElement).toHaveClass(
      checkboxClasses['color-green'],
    )
    expect(getByTestId('color-yellow').parentElement).toHaveClass(
      checkboxClasses['color-yellow'],
    )
    expect(getByTestId('color-green').parentElement).toHaveClass(
      checkboxClasses['color-green'],
    )
  })

  it('should add the appropriate size class to root element based on size prop', async () => {
    const { getByTestId } = await renderWithNexUIProvider(
      <>
        <Checkbox size='sm' data-testid='size-sm'>
          Checkbox
        </Checkbox>
        <Checkbox size='md' data-testid='size-md'>
          Checkbox
        </Checkbox>
        <Checkbox size='lg' data-testid='size-lg'>
          Checkbox
        </Checkbox>
      </>,
      {
        useAct: true,
      },
    )

    expect(getByTestId('size-sm').parentElement).toHaveClass(
      checkboxClasses['size-sm'],
    )
    expect(getByTestId('size-md').parentElement).toHaveClass(
      checkboxClasses['size-md'],
    )
    expect(getByTestId('size-lg').parentElement).toHaveClass(
      checkboxClasses['size-lg'],
    )
  })

  it('should add the appropriate radius class to root element based on radius prop', async () => {
    const { getByTestId } = await renderWithNexUIProvider(
      <>
        <Checkbox radius='sm' data-testid='radius-sm'>
          Checkbox
        </Checkbox>
        <Checkbox radius='md' data-testid='radius-md'>
          Checkbox
        </Checkbox>
        <Checkbox radius='lg' data-testid='radius-lg'>
          Checkbox
        </Checkbox>
        <Checkbox radius='full' data-testid='radius-full'>
          Checkbox
        </Checkbox>
      </>,
      {
        useAct: true,
      },
    )

    expect(getByTestId('radius-sm').parentElement).toHaveClass(
      checkboxClasses['radius-sm'],
    )
    expect(getByTestId('radius-md').parentElement).toHaveClass(
      checkboxClasses['radius-md'],
    )
    expect(getByTestId('radius-lg').parentElement).toHaveClass(
      checkboxClasses['radius-lg'],
    )
    expect(getByTestId('radius-full').parentElement).toHaveClass(
      checkboxClasses['radius-full'],
    )
  })

  it('should render unchecked checkbox by default', async () => {
    const { getByRole } = await renderWithNexUIProvider(<Checkbox />, {
      useAct: true,
    })
    const checkbox = getByRole('checkbox')
    expect(checkbox).not.toBeChecked()
  })

  it('should render checked checkbox by defaultChecked prop', async () => {
    const { getByRole } = await renderWithNexUIProvider(
      <Checkbox defaultChecked />,
      {
        useAct: true,
      },
    )
    const checkbox = getByRole('checkbox')
    expect(checkbox).toBeChecked()
  })

  it('should render checked checkbox when checked prop is true', async () => {
    const { getByRole, container } = await renderWithNexUIProvider(
      <Checkbox checked />,
      {
        useAct: true,
      },
    )
    const checkbox = getByRole('checkbox')
    expect(checkbox).toBeChecked()
    expect(container.firstElementChild).toHaveClass(checkboxClasses.checked)
  })

  it('should switch checked state and call onCheckedChange when clicked', async () => {
    const onCheckedChange = jest.fn()
    const { getByRole } = await renderWithNexUIProvider(
      <Checkbox onCheckedChange={onCheckedChange} />,
      {
        useAct: true,
      },
    )
    const checkbox = getByRole('checkbox')
    expect(checkbox).not.toBeChecked()

    await act(async () => {
      fireEvent.click(checkbox)
    })

    expect(checkbox).toBeChecked()
    expect(onCheckedChange).toHaveBeenCalledWith(true)

    await act(async () => {
      fireEvent.click(checkbox)
    })

    expect(checkbox).not.toBeChecked()
    expect(onCheckedChange).toHaveBeenCalledWith(false)
  })

  it('should forward classes to icon, root and label slots', async () => {
    const { container } = await renderWithNexUIProvider(
      <Checkbox
        classes={{
          icon: 'test-class-icon',
          root: 'test-class-root',
          label: 'test-class-label',
        }}
      >
        Checkbox
      </Checkbox>,
      {
        useAct: true,
      },
    )

    const root = container.querySelector(`.${checkboxClasses.root}`)
    const label = container.querySelector(`.${checkboxClasses.label}`)
    const icon = container.querySelector(`.${checkboxClasses.icon}`)

    expect(root).toHaveClass('test-class-root')
    expect(icon).toHaveClass('test-class-icon')
    expect(label).toHaveClass('test-class-label')
  })

  it('should forward slotProps to icon, root and label slots', async () => {
    const { container } = await renderWithNexUIProvider(
      <Checkbox
        slotProps={{
          root: { className: 'test-class-root' },
          label: { className: 'test-class-label' },
          icon: { className: 'test-class-icon' },
        }}
      >
        Checkbox
      </Checkbox>,
      {
        useAct: true,
      },
    )

    const root = container.querySelector(`.${checkboxClasses.root}`)
    const label = container.querySelector(`.${checkboxClasses.label}`)
    const icon = container.querySelector(`.${checkboxClasses.icon}`)

    expect(root).toHaveClass('test-class-root')
    expect(label).toHaveClass('test-class-label')
    expect(icon).toHaveClass('test-class-icon')
  })

  it('should disable the checkbox when disabled prop is true', async () => {
    const onCheckedChange = jest.fn(() => {})

    const { getByRole, container } = await renderWithNexUIProvider(
      <Checkbox disabled onCheckedChange={onCheckedChange} />,
      {
        useAct: true,
      },
    )

    const checkbox = getByRole('checkbox')

    expect(checkbox).toBeDisabled()
    expect(container.firstElementChild).toHaveClass(checkboxClasses.disabled)
    await act(async () => {
      fireEvent.click(checkbox)
    })
    expect(onCheckedChange).not.toHaveBeenCalled()
  })

  it('should render indeterminate checkbox when indeterminate prop is true', async () => {
    const { getByRole, container } = await renderWithNexUIProvider(
      <Checkbox indeterminate defaultChecked />,
      {
        useAct: true,
      },
    )
    const checkbox = getByRole('checkbox')
    expect(checkbox).toBeInTheDocument()
    expect(checkbox).toBeChecked()
    expect(container.firstElementChild).toHaveClass(
      checkboxClasses.indeterminate,
    )
  })

  it('should render custom icon when icon prop is provided', async () => {
    const { getByTestId, rerender, container } = await renderWithNexUIProvider(
      <Checkbox
        defaultChecked
        icon={<span data-testid='custom-icon'>Icon</span>}
      />,
      {
        useAct: true,
      },
    )
    const icon = getByTestId('custom-icon')
    expect(icon).toBeInTheDocument()

    const renderIcon = jest.fn(() => (
      <span data-testid='custom-icon'>Icon</span>
    ))

    rerender(<Checkbox icon={renderIcon} defaultChecked />)

    expect(renderIcon).toHaveBeenCalledWith({
      type: 'checkbox',
      checked: true,
      disabled: false,
      size: 'md',
      radius: 'md',
      color: 'blue',
      defaultChecked: true,
      inGroup: false,
      name: undefined,
      icon: renderIcon,
    })
    expect(getByTestId('custom-icon')).toBeInTheDocument()

    rerender(<Checkbox icon='123' defaultChecked />)

    expect(
      container.querySelector(`.${checkboxClasses.icon}`),
    ).toHaveTextContent('123')
  })

  it("should warn when Checkbox's checked is set within CheckboxGroup", async () => {
    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation()

    await renderWithNexUIProvider(
      <CheckboxGroup>
        <Checkbox value='orange' checked>
          orange
        </Checkbox>
      </CheckboxGroup>,
      {
        useAct: true,
      },
    )

    expect(consoleSpy).toHaveBeenCalledWith(
      '[Nex UI] Checkbox: The CheckboxGroup is being used, `checked` will be ignored. Use the `value` of the CheckboxGroup instead.',
    )

    consoleSpy.mockRestore()
  })

  it("should warn when Checkbox's defaultChecked is set within CheckboxGroup", async () => {
    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation()

    await renderWithNexUIProvider(
      <CheckboxGroup>
        <Checkbox value='orange' defaultChecked>
          orange
        </Checkbox>
      </CheckboxGroup>,
      {
        useAct: true,
      },
    )

    expect(consoleSpy).toHaveBeenCalledWith(
      '[Nex UI] Checkbox: The CheckboxGroup is being used, `defaultChecked` will be ignored. Use the `defaultValue` of the CheckboxGroup instead.',
    )

    consoleSpy.mockRestore()
  })

  it("should warn when Checkbox's value is not set within CheckboxGroup", async () => {
    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation()

    await renderWithNexUIProvider(
      <CheckboxGroup>
        <Checkbox>orange</Checkbox>
      </CheckboxGroup>,
      {
        useAct: true,
      },
    )

    expect(consoleSpy).toHaveBeenCalledWith(
      '[Nex UI] Checkbox: The CheckboxGroup is being used, but `value` is not provided.',
    )

    consoleSpy.mockRestore()
  })
})
