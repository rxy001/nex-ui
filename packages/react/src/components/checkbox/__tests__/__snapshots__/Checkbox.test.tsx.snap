// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render with default props 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  height: var(--nui-sizes-8);
  padding: var(--nui-spaces-1-5);
}

.emotion-1 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 0;
  box-shadow: none;
  box-sizing: border-box;
  position: absolute;
  inset: 0;
  opacity: 0;
  cursor: inherit;
  z-index: 1;
}

.emotion-1[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}

.emotion-1[type="search"]::-webkit-search-cancel-button {
  -webkit-appearance: none;
}

.emotion-1:is(:-webkit-autofill, :autofill) {
  background-color: transparent!important;
  -webkit-transition: background-color 50000s ease-in-out 0s;
  transition: background-color 50000s ease-in-out 0s;
}

.emotion-1[data-focus-visible=true]:not(:disabled, [data-disabled=true])+*::before {
  outline: var(--nui-borders-md) var(--nui-colors-blue-primary);
  outline-offset: var(--nui-spaces-0-5);
}

.emotion-1:is(:hover, [data-hover=true]):not(:disabled, [data-disabled=true])+*::before {
  border-color: var(--nui-colors-gray-secondary);
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--nui-colors-blue-contrast-text);
  width: var(--nui-sizes-5);
  height: var(--nui-sizes-5);
  font-size: var(--nui-font-sizes-md);
}

.emotion-2::before {
  content: "";
  position: absolute;
  inset: 0;
  border: var(--nui-borders-md);
  border-color: var(--nui-colors-gray-highlight);
  -webkit-transition: var(--nui-transitions-colors);
  transition: var(--nui-transitions-colors);
  background: transparent;
  z-index: -2;
  border-radius: calc(var(--nui-radii-lg) * .6);
}

.emotion-2::after {
  content: "";
  position: absolute;
  inset: 0;
  background-color: var(--nui-colors-blue-primary);
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  -webkit-transform: scale(0.4);
  -moz-transform: scale(0.4);
  -ms-transform: scale(0.4);
  transform: scale(0.4);
  opacity: 0;
  z-index: -1;
  border-radius: calc(var(--nui-radii-lg) * .6);
}

.emotion-3 {
  -webkit-margin-start: var(--nui-spaces-2);
  margin-inline-start: var(--nui-spaces-2);
  font-size: var(--nui-font-sizes-lg);
}

<label
  class="nui-checkbox-root nui-checkbox-radius-md nui-checkbox-size-md nui-checkbox-color-blue emotion-0"
>
  <input
    aria-label="Checkbox"
    aria-labelledby="«rj»"
    class="nui-checkbox-input emotion-1"
    tabindex="0"
    type="checkbox"
  />
  <span
    class="nui-checkbox-icon emotion-2"
  >
    <svg
      aria-hidden="true"
      fill="none"
      focusable="false"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m7 12.9l3.143 3.6L18 7.5"
        opacity="0"
        pathLength="1"
        stroke-dasharray="0px 1px"
        stroke-dashoffset="0px"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </span>
  <span
    class="nui-checkbox-label emotion-3"
    id="«rj»"
  >
    Checkbox
  </span>
</label>
`;
